lib/c/c_ClientConfiguration.cc

```diff
@@ -78,7 +78,7 @@ class PulsarCLogger : public pulsar::Logger {
     PulsarCLogger(const std::string &file, pulsar_logger logger, void *ctx)
         : file_(file), logger_(logger), ctx_(ctx) {}
 
-    bool isEnabled(Level level) { return level >= pulsar::Logger::LEVEL_INFO; }
+    bool isEnabled(Level level) { return true; }
```

Comment:
Could we support setting the level in the PulsarCLogger. Like SimpleLogger here: https://github.com/apache/pulsar-client-cpp/blob/5d5ed2acc2e89aed90b593343369f360644440e7/lib/SimpleLogger.h#L53
Otherwise, all debug output-related calculations will be called, which may have some impact on the performance.

Reply:
For C users, We expose a method: `pulsar_logger `.

https://github.com/apache/pulsar-client-cpp/blob/0e86e9f1f8304fd0bbf21185c1c66d0b5ac58482/include/pulsar/c/client_configuration.h#L36-L37

`True` is always returned here in order to let user controls what level of the log is printed in this method.

Such as:

``` c
void custom_logger(pulsar_logger_level_t level, const char *file, int line, const char *message,
              void *ctx) {
    // Control the log level yourself.
    if (level >= pulsar_INFO) {
         printf("[%u] [%s] [%d] [%s] \n", level, file, line, message);
    }
}
```

This approach avoids the need to expose additional interfaces to set the log level.

Reply:
Take this as an example:
https://github.com/apache/pulsar-client-cpp/blob/5d5ed2acc2e89aed90b593343369f360644440e7/lib/MessageCrypto.cc#L153-L157
In this case, lines 154-156 will always be executed even though the user doesn't want to get the debug log. We usually use this approach to avoid executing some performance-critical code when we don't need to print the debug log.

Reply:
Good point. I will change the solution.

Reply:
@RobertIndie I added `pulsar_client_configuration_set_logger_and_level` to  change log level, PTAL.